tests/functional/test_kinesis.py

```diff
@@ -0,0 +1,92 @@
+    def setUp(self):
+        super(TestKinesisListStreams, self).setUp()
+        self.stream_name = "kinesis-test-stream"
+        self.region = "us-east-1"
+        self.client = self.session.create_client("kinesis", self.region)
+        self.http_stubber = ClientHTTPStubber(self.client)
+        self.http_stubber.add_response()
+
+    def test_can_put_stream_blob(self):
+        unique_data = str(uuid4())
+        with self.http_stubber as stub:
+            self.client.put_record(
+                StreamName=self.stream_name, PartitionKey="foo", Data=unique_data
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(request["StreamName"], self.stream_name)
+            self.assertEqual(request["PartitionKey"], "foo")
+            self.assertEqual(
+                b64decode(request["Data"]).decode("utf-8"),
+                unique_data
+            )
+
+    def test_can_put_records_single_blob(self):
+        unique_data = str(uuid4())
+        with self.http_stubber as stub:
+            self.client.put_records(
+                StreamName=self.stream_name,
+                Records=[{"Data": unique_data, "PartitionKey": "foo"}],
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(len(request["Records"]), 1)
+            self.assertEqual(request["StreamName"], self.stream_name)
+
+            record = request["Records"][0]
+            self.assertEqual(record["PartitionKey"], "foo")
+            self.assertEqual(
+                b64decode(record["Data"]).decode("utf-8"),
+                unique_data
+            )
+
+    def test_can_put_records_multiple_blob(self):
+        with self.http_stubber as stub:
+            self.client.put_records(
+                StreamName=self.stream_name,
+                Records=[
+                    {"Data": "foobar", "PartitionKey": "foo"},
+                    {"Data": "barfoo", "PartitionKey": "foo"},
+                ],
+            )
+            self.assertEqual(len(stub.requests), 1)
+            # TODO: Decode for 3.4 and 3.5 (remove after deprecation)
+            request = json.loads(stub.requests[0].body.decode("utf-8"))
+            self.assertEqual(len(request["Records"]), 2)
+
+            record_foobar = request["Records"][0]
+            record_barfoo = request["Records"][1]
+            self.assertEqual(
+                b64decode(record_foobar["Data"]).decode("utf-8"),
+                "foobar"
+            )
+            self.assertEqual(
+                b64decode(record_barfoo["Data"]).decode("utf-8"),
+                "barfoo"
+            )
+
+
+if __name__ == "__main__":
```

Comment:
Not really a problem, but we don't necessarily need the `if` statement here as we really only run the tests with `nose`.